src/metadata.c

```diff
@@ -28,6 +28,9 @@ vcsbeam_context *vmInit( bool use_mpi )
     // Allocate memory for the VCSBEAM_METADATA struct
     vcsbeam_context *vm = (vcsbeam_context *)malloc( sizeof(vcsbeam_context) );
 
+    gpuSetDevice(0);
```

Comment:
Were these additions necessary for things to run correctly? What about when there's more than a single GPU per device (e.g., on a Setonix gpu node)?

Reply:
No, in fact, I revoke my claim that adding this fixes anything at all. After rerunning on my computer several times (in the course of debugging), I found that it started randomly failing _despite_ this addition. So I do *not* recommend that this commit gets included.

FWIW, I later found evidence (again, on my home computer) that there was a persistent memory leak that slowly ate up more and more memory which wasn't released when the code terminated. I now think that the intermittent failures were related to this memory leak---in particular, when my (CPU) memory was almost "full". I couldn't figure out how to release said memory except by restarting my machine.

Reply:
Oh gosh, okay.